### PR TITLE
Update report messages to include function name

### PR DESCRIPTION
**Pull request by Moeriki**
Fix for #7260 (discussion pending)

### CODE REVIEW

**Comment by vitorbal** [comment]
Thanks, @Moeriki! Adding "do not merge" for now as the corresponding issue is not marked as accepted yet.

**Comment by platinumazure** [comment]
@Moeriki If it's not too much trouble, could you please answer the questions in the pull request template, just so we know what I's you have dotted and T's you have crossed? Thanks!

I've reproduced the template below for your convenience.

```
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix ([template](https://github.com/eslint/eslint/blob/master/templates/bug-report.md))
[ ] New rule ([template](https://github.com/eslint/eslint/blob/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://github.com/eslint/eslint/blob/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

<!--
    If the item you've checked above has a template, please paste the template questions below and answer them. (If this pull request is addressing an issue, you can just paste a link to the issue here instead.)
-->


<!--
    The following is required for all pull requests:
-->

**Please check each item to ensure your pull request is ready:**

- [ ] I've read the [pull request guide](http://eslint.org/docs/developer-guide/contributing/pull-requests)
- [ ] I've included tests for my change
- [ ] I've updated documentation for my change (if appropriate)

**What changes did you make? (Give an overview)**


**Is there anything you'd like reviewers to focus on?**
```

**Comment by Moeriki** [comment]
I'll update Pull Request to filled out template once #7260 is completely analysed.

**Comment by nzakas** [comment]
The issue was accepted, so I removed "do not merge."

File: tests/lib/rules/max-statements.js [hunk 1]
```diff
@@ -34,20 +34,20 @@ ruleTester.run("max-statements", rule, {
...
-        { code: "define(['foo', 'qux'], function(foo, qux) { var bar = 1; var baz = 2; return function () { var z; return 42; }; })", options: [1, {ignoreTopLevelFunctions: true}], errors: [{ message: "This function has too many statements (2). Maximum allowed is 1."}] },
-        { code: "function foo() { var a; var b; var c; var x; var y; var z; bar(); baz(); qux(); quxx(); foo(); }", errors: [{ message: "This function has too many statements (11). Maximum allowed is 10."}] },
+        { code: "function foo() { var bar = 1; var baz = 2; var qux = 3; }", options: [2], errors: [{ message: "Function 'foo' has too many statements (3). Maximum allowed is 2."}] },
+        { code: "var foo = () => { var bar = 1; var baz = 2; var qux = 3; };", options: [2], parserOptions: { ecmaVersion: 6 }, errors: [{ message: "Function 'anonymous' has too many statements (3). Maximum allowed is 2."}] },
```
Comment (nzakas) [inline tests/lib/rules/max-statements.js]: Based on the comments on #7260, it looks like the decision was to use "this function" in the case of anonymous functions. Can you update that?

**Comment by nzakas** [comment]
@Moeriki are you planning on updating this PR?

**Comment by Moeriki** [comment]
I was working on it. Bit busy now. I hope to continue and finish within a few weeks.

**Comment by nzakas** [comment]
@Moeriki thanks for the fast response. One more question: would you be okay with someone finishing this up for you? It's already been open for 21 days, and it would be nice to finish it up sooner rather than later.

**Comment by Moeriki** [comment]
Sure if anyone who likes to do it np.

**Comment by nzakas** [comment]
Cool, I'll take it over.

**Comment by nzakas** [comment]
Closing in favor of https://github.com/eslint/eslint/pull/7399

**Pull request closed without merging**